skore/src/skore/project/_launch.py

```diff
@@ -0,0 +1,183 @@
+                self._server_running = False
+                try:
+                    tasks = asyncio.all_tasks(self._loop)
+                    for task in tasks:
+                        task.cancel()
+                    self._loop.run_until_complete(
+                        asyncio.gather(*tasks, return_exceptions=True)
+                    )
+                    self._loop.close()
+                except Exception:
+                    pass
+                console.rule("[bold cyan]skore-UI[/bold cyan]")
+                console.print(
+                    f"Server that was running at http://localhost:{self._port} has "
+                    "been closed"
+                )
+
+        self._executor = ThreadPoolExecutor(max_workers=1)
+        self._executor.submit(run_in_thread)
+        project._server_manager = self
+        self._executor.shutdown(wait=False)
+
+    def shutdown(self):
+        """Shutdown the server and cleanup resources."""
+        if self._executor is not None and self._server_running:
+            if self._loop is not None:
+
+                async def stop():
+                    tasks = asyncio.all_tasks(self._loop)
+                    for task in tasks:
+                        if not task.done():
+                            task.cancel()
+                    await asyncio.gather(*tasks, return_exceptions=True)
+
+                try:
+                    future = asyncio.run_coroutine_threadsafe(stop(), self._loop)
+                    future.result(timeout=0.1)
+                except Exception:
+                    pass
+
+            self._server_running = False
+            with contextlib.suppress(Exception):
+                self._executor.shutdown(wait=True, cancel_futures=True)
+            self._executor = None
+            self._loop = None
+
+
+async def run_server(project: Project, port: int, open_browser: bool, console: Console):
+    @asynccontextmanager
+    async def lifespan(app: FastAPI):
+        if open_browser:
+            webbrowser.open(f"http://localhost:{port}")
+        try:
+            yield
+        except asyncio.CancelledError:
+            project._server_manager = None
+
+    from skore.ui.app import create_app
+
+    app = create_app(project=project, lifespan=lifespan)
+    server_config = uvicorn.Config(
+        app,
+        port=port,
+        log_level="error",
+        log_config={
+            "version": 1,
+            "disable_existing_loggers": True,
+            "handlers": {
+                "default": {
+                    "class": "logging.NullHandler",
+                }
+            },
+            "loggers": {
+                "uvicorn": {"handlers": ["default"], "level": "ERROR"},
+                "uvicorn.error": {"handlers": ["default"], "level": "ERROR"},
+            },
+        },
+    )
+    server = uvicorn.Server(server_config)
+    console.print(
+        f"Running skore UI from '{project.name}' at URL http://localhost:{port}"
```

Comment:
```suggestion
        f"Running skore UI from '{project.name}' at URL http://localhost:{port}"
        "Stop your kernel or hit ctrl + C in your terminal to stop it."
```

If you run a `skore.open()` in a regular script, the script never finish (as the server is running) which is fine. But user needs to hit ctrl+C to exit maybe this could be explain it in the print.